source/GStreamerEMEUtils.cpp

```diff
@@ -125,6 +125,59 @@ void getInitWithLast15FromProtectionMetadata(GstRialtoProtectionMetadata *protec
     metadata.initWithLast15 = initWithLast15;
 }
 
+void getEncryptionSchemeFromProtectionMetadata(GstRialtoProtectionMetadata *protectionMeta,
+                                               BufferProtectionMetadata &metadata)
+{
+    const char *cipherModeBuf = gst_structure_get_string(protectionMeta->info, "cipher-mode");
+    GST_INFO("Retrieved encryption scheme '%s' from protection metadata.", cipherModeBuf ? cipherModeBuf : "unknown");
+    if (g_strcmp0(cipherModeBuf, "cbcs") == 0)
+    {
+        metadata.cipherMode = firebolt::rialto::CipherMode::CBCS;
+    }
+    else if (g_strcmp0(cipherModeBuf, "cenc") == 0)
+    {
+        metadata.cipherMode = firebolt::rialto::CipherMode::CENC;
+    }
+    else if (g_strcmp0(cipherModeBuf, "cbc1") == 0)
+    {
+        metadata.cipherMode = firebolt::rialto::CipherMode::CBC1;
+    }
+    else if (g_strcmp0(cipherModeBuf, "cens") == 0)
+    {
+        metadata.cipherMode = firebolt::rialto::CipherMode::CENS;
+    }
+    else
+    {
+        if (cipherModeBuf)
```

Comment:
I think its better if rialto-gstreamer is dumb and just passes through the values to rialto, so I would just not set or log anything if cipherModeBuf isnt found. RialtoServer can handle any errors this causes.